Assignment6.Rmd

```diff
@@ -22,24 +22,25 @@ library(rpart)
 ```{r}
 #Upload the data sets MOOC1.csv and MOOC2.csv
 M1 <- read.csv("MOOC1.csv", header = TRUE)
-
-M2 <- 
-
+View(M1)
+M2 <- read.csv("MOOC2.csv", header = TRUE)
+View(M2)
 ```
 
 #Decision tree
 ```{r}
 #Using the rpart package generate a classification tree predicting certified from the other variables in the M1 data frame. Which variables should you use?
 
-c.tree1 <- 
+# Below I use ctree to construct a model of certification as a function of all other variables, including forum posts, grades and assignments. Optional parameters are integrated for controlling tree growth. control=rpart.control(minsplit=1, cp=0.001) requires that the minimum number of observations in a node be 1 before attempting a split and that a split must decrease the overall lack of fit by a cost complexity factor of 0.001 before being attempted.
+c.tree1 <- rpart(certified ~ forum.posts+grade+assignment, data=M1, method = "class", control=rpart.control(minsplit=1, minbucket = 1, cp=0.001))
```

Comment:
Good work 👍 You should `scale()` the variables before putting them in the model though, otherwise `forum.posts` tends to swamp the other variables as it has many more increments.